### ISSUE TITLE
Suddenly the buttons can no longer perform their actions, they aren't starting the action flow properly

### ISSUE THREAD

**Issue by apppbuilder** [FlutterFlow/flutterflow-issues#4048]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Some buttons in my project are not working as expected anymore.

They used to worked very well in older versions, the last bundle was sent for production 5 Sept 2024 in Play Store.

Maybe some update from the FF Team changed some code flow and it's not performing anymore.

I did lot of tests and found what was happening, but can't fix it because it's not an action flow issue, but maybe a structured code issue...

In the process to discover what was happening, I created an App State called "errorState" (string), and after each action I would update it with a new string to see where it stopped, but after testing every possibility, I noticed that the problem was not in the action flow, but the button WAS NOT EVEN STARTING the action flow.

At this moment, after clicking the button, it just break the app, freezing the screen and the only way to get it back alive is by closing and opening the app again...

I don't know how to reproduce it in a new project because the button has lot of actions inside and it was working VERY WELL 3 months ago, so I would need a project from that date to check both together.

All the tests were made in internal testing through Google Play Store.

**IMPORTANT:** One thing I noticed is that the error occurs only in the BUTTON WIDGET. Some times I like to use a manual built button with container and text widgets, it's working really well in this case, but with the BUTTON WIDGET, nothing is working.

**IMPORTANT:** Another button that isn't working is one created through custom code, I'll link it in the bug repport below.

### Expected Behavior

The button should start the action flow properly.

### Steps to Reproduce

1. Widget button in the screen.
2. Set the action flow inside it.
3. When clicking it, the button does not start the action flow.

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

Button 1: IT4CjPLlvJFIocte7q6JcMdVoCMvNlw2RYIw1+xEVg8cDZPOD7AQP8nRP1dUOM/iSnNmfGD8nnwX/MrRhezXGPBdBzmCGdh81LViZBXyLVimSpDXC763b3J8M/1mJlSC58K3phIkOvRoL2NgwEylNdmtUhbXH4DzC1g3NOa4XdeK2SrDX1iXc2URm05KZDPz // Custom Code Button 2: IT4CjPLlvJFIocte7q6JcMdVoCMvNlw2RYIw1+xEVg8cDZPND+x/esnRQFJuZ9OrdglYOGaGoH8e/vDXm9z1NPoECzKqbtU7wbV6ajuXJX+Vbp+VBs6NdEBUGshVJmKg4ZiRnSUkOs1uLkIi3Tu1e/O9QT3qJKLTNmwjBs+XIqnXvgfuQ0SLb3kNh1JWeC/v

### Visual documentation

This is the "device 1".

Steps:
1. Clicked the image "História" and it was led to selectHistoryStud page.
2. In this new page clicked in the "filter button" and in "estilo button" built with conteiners in the page. Everything working well.
3. After clicking in the button "Começar História" (a button widget), the screen freezes and nothing works anymore.

PS.: In this recording I pressed and holded the button, but it was just to show the time I was clicking in it, but even with a simple click it just stop everything.

https://github.com/user-attachments/assets/dd4fee6b-afb4-459f-bb89-4b6f5ba92656

In the custom code button the thing is almost the same. But somehow in another device I'm able to go further the button "Começar história" but can't do simple actions inside the new screen with a built button "Aperte para Falar".

In this case, the only difference is that the screen doesn't freeze.

https://github.com/user-attachments/assets/c4560b48-ccd3-482d-b7a6-13cb83924345

### Environment

```markdown
- FlutterFlow version: FlutterFlow v5.0.6+ released October 12, 2024. Flutter version is 3.22.2.
- Platform: Website.
- Browser name and version: Google Chrome 129.0.6668.90
- Operating system and version affected: Windows 11, Android not working properly
```

### Additional Information

The issue completely stopped the app to work properly. Users are unable to perform simple actions that are crucial in the app functionality.

**Comment by Alezanello** [comment]
Hi there!

I was reviewing your code and noticed that you have several "Navigate To" actions followed by Custom Actions. This can lead to performance issues because the code attempts to execute actions after navigation, even though the previous page may already be disposed of, which can cause errors.

I strongly recommend adding a "Terminate Action" after the navigation, or moving the Custom Actions to run before navigating to the new page.

Or delete all navigates to and have it at the end of all the flow and before the navigate to trigger the custom action you have.

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/d9e1eeb0-57fe-49e0-b24a-a03115c55161/41ff0781-09c8-419b-b411-f6a8bdc89af7?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS9kOWUxZWViMC01N2ZlLTQ5ZTAtYjI0YS1hMDMxMTVjNTUxNjEvNDFmZjA3ODEtMDljOC00MTliLWI0MTEtZjZhOGJkYzg5YWY3IiwiaWF0IjoxNzI4OTA4MDE5LCJleHAiOjMzMjk5NDY4MDE5fQ._CZY6AIcltc4HwLeQzMv1vdnUhvEUiX9-iSDd6WS9z4 " alt="image.png" width="1157" height="738" />

**Comment by apppbuilder** [comment]
Hey @Alezanello

Thanks for replying. I did not only what you mentioned, but also a test to check if my hypothesis is correct.

1. Did everything you mentioned about adding terminate in the button action flow.
2. Created another button below the native button widget with conteiner and text. Same actions are used in both with the only difference on the output variable names on the API Calls.

**Result:**

Ok, now I'm scared. None of the buttons worked. Both froze the screen and I needed to close the app...

**Comment by apppbuilder** [comment]
I don't know if it helps, but I have another page, used in tutorial/first login only, it's almost the same, but with much less actions in the flow and it's working good...

Maybe something is going on having much actions... I have no idea...

Here's the code bug report to check this tutorial button:

IT4CjPLlvJFIocte7q6JcMdVoCMvNlw2RYIw1+xEVg8cDZPOD7AQP8nRP1dUOc/iSm9iO2f9rGQX/MnIuN7AU/oEYD6ZQIs61rVbWwnKRka7RaaVBa63b3J8M/1mJlSC58K3phIkOvRoL2NgwEylNdmtUhbXH4DzC1g3NOa4XdeK2SrDX1iXc2URm05KZDPz

The action flow in this case starts with an update app state. Already made a test starting the "normal" button with an action of "update app state", but nothing new...

**Comment by sanjayelate** [comment]
I could be wrong but these issues all might be related:

https://github.com/FlutterFlow/flutterflow-issues/issues/4000
https://github.com/FlutterFlow/flutterflow-issues/issues/4001
https://github.com/FlutterFlow/flutterflow-issues/issues/4047
https://github.com/FlutterFlow/flutterflow-issues/issues/4048

**Comment by apppbuilder** [comment]
Hello @sanjayelate , thanks for replying.

If it helps, I'll describe a little more the project issue:

1. Every API Call used in this action flow calls a Cloud Function in Google Cloud (and no, I'll not switch to native Cloud Functions tab on FF, already lost almost 2 weeks trying to configure it one time but there was almost no data/support at that time to help, and **it's working very good as API Calls**, always worked).

2. This is one of the biggest action flows used in my app (24 actions). In the home page I have 44 actions triggered On Page Load and everything is ok there.

3. After clicking the button, the screen freezes and no action is initiated. Again, **no action is initiated**. I know it because I created an "update app state" on a App State String called "erroState" and updated before and after every action. When clicking the button, the screen freezes completely and no update is seem in the "erroState" App State. (maybe initiating, but can't be seen because the app brokened completely with screen freezed?)

4. I have also another custom code that used to work very well, no changes were made in it, and when I click the button with a OnPress action, it stops working, but in this case I can still click in the things that appear in the screen.

5. Every error I mentioned above was saw in internal testing on Android Devices (after deploying to PlayStore). They used to work very well in the past, but now they are broken...

My thoughts:

1. Maybe it's limiting by the amount of actions? But the custom code doesn't have much actions...
2. Maybe it's a certain type of action that is breaking the code?
3. Maybe some brackets are being eliminated or not being used properly after/before deployment and the whole action flow becomes broken?

**Comment by apppbuilder** [comment]
@Alezanello any point on this? The app is still broken and I can't go through with it...

**Comment by apppbuilder** [comment]
Hello @Alezanello , is there any update or anything I can do to help fixing it?

**Comment by Alezanello** [comment]
Hello, and I apologize for the delayed response. After reviewing your code, I still don't see any issues. One thing we can do to help debug this is to log specific points in the action flow to see where it stops working.

To do this, you can create a simple custom action with a string argument named `Log` and use the following code:

```dart
Future printConsoleLog(String log) async { 
print(log); 
}
```

You can trigger this custom action before each step in your action flow. In the `Log` argument, pass a message indicating which part of the flow you're at. This will help us pinpoint where the issue occurs and make it easier to debug the error.

Let me know how it goes or if you need further assistance!

**Comment by apppbuilder** [comment]
@Alezanello after lots of testing, found out what was happening. There was an infinite loop inside a custom function that happened only if a certain condition were met inside the app... that's why it was ok inside a smartphone and in another one it was breaking...

Also, about the custom code that I said before, we found out that there was a bug inside the code, we solved it by flaging every action inside an app state that showed in the screen.

Thank you very much for the patience and directions!
